specific_use_cases/DFU_work_in_progress/station_8760s.ipynb

```diff
@@ -0,0 +1,627 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "47d1abf5-1991-4fd9-8344-af5d7925e3ab",
+   "metadata": {},
+   "source": [
+    "We could stop there, but let's do a little more. Now that we have hourly data to work with, we can *improve upon the constructed 8760's that had otherwise been in use!*\n",
+    "\n",
+    "#### Sample the \"8760\" closest to the median\n",
+    "\n",
+    "One way that you may wish to use this data is to examine \"8760's\", or entire years of hourly data. If you chose a wide range of years previously, you might want to now sub-select, for example, only a historical reference period."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "61cebe21-8afb-4dd3-8e4c-80f2d1e357a9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = data.sel(time=slice('19810101','20101231'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b352904-8ab5-4767-9d04-52fab3b7b1fb",
+   "metadata": {},
+   "source": [
+    "Beware that if you selected 'historical climate', and don't do something like the step above you will have partial years at the beginning and end, since the simulations start in Sept and run through Aug of the following calendar year. You could alternatively deal with that this way:"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "id": "4f70f7e5-2525-4009-8cd1-dc045c10eaf9",
+   "metadata": {},
+   "source": [
+    "min_year = data['time.year'].min().values\n",
+    "max_year = data['time.year'].max().values\n",
+    "data = data.sel(time=slice(str(min_year+1)+'0101',str(max_year-1)+'1231'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5d8fd82f-8b82-4394-883a-e909025145bf",
+   "metadata": {},
+   "source": [
+    "How many 8760s are there in this dataset for each station location?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a2c91a8-4a0d-4a33-a88d-ec02241921d5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "number_of_years = (data['time.year'].max() - data['time.year'].min()).values\n",
+    "unique_8760s = number_of_years * len(data.simulation) \n",
+    "unique_8760s"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc351b96-1ce8-4338-a66b-5c33d71aa2d1",
+   "metadata": {},
+   "source": [
+    "For each station, let's find the \"8760\" that is closest to the median (the euclidean distance between that year and the median across all days and hours is smallest).\n",
+    "\n",
+    "Prepare helper functions. (You could also loop over days and times, but that would take much longer to execute.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "03bfdd3b-9fa3-46fe-b134-0ba4361ee826",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import xarray as xr\n",
+    "\n",
+    "def get_difference_hour(y): \n",
+    "    #for this hour, of this day of the year, the median across all 8760s:\n",
+    "    median = y.median()\n",
```

Comment:
Unsure how easy this is to change out (and/or if this is a direction we want to go in at all), but it would be nice if this was a quantile function (and we could still show an example w/ 0.5 as the quantile), so the method was flexible and could pull e.g. a hot year for some percentile threshold

Reply:
Not at all hard to replace .median() with .quantile(q=0.5), which may make it clearer how to modify this. (Also added a note that one could modify that.)